api/books/markAsRead.ts

```diff
@@ -0,0 +1,23 @@
+import { baseWithAuth } from "api/base"
+import { BookType } from "api/types"
+
+export const markAsRead = async (id: number): Promise<BookType> => {
+  return await baseWithAuth()
+    .request({
+      url: `/api/user_books/${id}`,
+      method: "PATCH",
+      data: {
+        user_book: {
+          finish_date: new Date().toISOString(),
+        },
+      },
+    })
+    .then((res) => {
+      const book = res.data
+      console.log(res)
+      return book
+    })
+    .catch((error) => {
+      console.log(error)
```

Comment:
Can you return a proper error response?

Reply:
right. I'll go ahead and make a seperate PR for the error handling. I've been putting it aside because I wasn't so sure how to do those yet but its about time I did them. yeah